### DISCUSSION TITLE
Is AWS SDK for c++ Y2k38 safe?

### DISCUSSION THREAD

**Discussion by bzhou-sw** [aws/aws-sdk-cpp#3050] [Q&A] [answered]
Year 2038 is not too far away, as a SDK user, we want to know if AWS SDK for c++ Y2k38 safe. Thanks a lot.

**Comment by jmklix** [comment, marked as answer]
C++ SDK uses C++ `std::chrono::system_clock::time_point` class to represent timepoint. If your compiler is using y2k38 safe implementation of this class - the SDK itself is safe. However, SDK also uses a bunch of other dependencies such as libCrypto (OpenSSL by default), libCurl, cJson, tinyXML, etc.... we are not sure if they are y2k38 safe

**Reply by bzhou-sw** [reply]
Thanks!
